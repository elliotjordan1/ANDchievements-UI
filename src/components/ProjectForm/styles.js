import styled from 'styled-components';
import theme from '../../global/theme';

export const FormTitle = styled.div`
    background-color: black;
    color: white;
    font: 500 24px ${theme.fontFamily.primary};
    line-height: 1.4;
    letter-spacing: 2px;
    width: 100%;
    text-indent: 8px;
    height: min-content;
    @media ${theme.media.tablet} {
        font-size: 32px;
    };
    @media ${theme.media.desktop} {
        font-size: 45px;
    };
`;

export const FormBody = styled.div`
    background-color: white;
    margin: 8px auto;
    padding: 8px 8px 24px 8px;
    @media ${theme.media.tablet} {
        padding: unset;
        width: 540px;
        min-height: 50vh;
        margin: 16px auto;
    };
    @media ${theme.media.desktop} {
        width: 634px;
        margin: 32px auto;
    };
`;

export const Label = styled.div`
    font: 500 16px ${theme.fontFamily.primary};
    letter-spacing: 1px;
    @media ${theme.media.tablet} {
        font-size: 18px;
    };
    @media ${theme.media.desktop} {
        font-size: 20px;
    };
`;

export const FormInput = styled.input`
    width: 100%;
    padding: 0 8px;
    box-sizing: border-box;
    border:  1px solid black;
    margin: 8px 0;
    font: 300 14px ${theme.fontFamily.primary};
    height: 40px;
    @media ${theme.media.tablet} {
        font-size: 16px;
        margin: 8px 0 16px 0;
        height: 40px;
        padding-left: 16px;
    };
    @media ${theme.media.desktop} {
        font-size: 18px;
        height: 50px;
        margin: 8px 0 24px 0;
    };
`;

export const TextArea = styled.textarea`
    width: 100%;
    padding: 4px 8px;
    box-sizing: border-box;
    resize: none;
    border: none;
    font: 300 14px ${theme.fontFamily.primary};
    height: 41px;
    margin: none;
    @media ${theme.media.tablet} {
        font-size: 16px;
        height: 51px;
        padding: 4px 16px;
    };
    @media ${theme.media.desktop} {
        font-size: 18px;
        height: 51px;
    };
`;

export const InputContainer = styled.div`
    border: 1px solid black;
    margin: 8px 0;
    @media ${theme.media.tablet} {
        margin: 8px 0 16px 0;
    };
    @media ${theme.media.desktop} {
        margin: 8px 0 24px 0;
    };
`;

export const SubmitButton = styled.button`
    background-color: ${theme.colours.brand.red};
    width: 120px;
    height: 40px;
    display: block;
    margin: 8px auto;
    border: none;
    color: white;
    font: 500 14px ${theme.fontFamily.primary};
    letter-spacing: 2px;
    margin-top: 8px;
    @media ${theme.media.tablet} {
        width: 160px;
        height: 50px;
        font-size: 16px;
        letter-spacing: 2px;
        margin-top: 16px;
    };
    @media ${theme.media.desktop} {
        width: 200px;
        height: 60px;
        font-size: 18px;
        letter-spacing: 3px;
        margin-top: 24px;
    };
`;