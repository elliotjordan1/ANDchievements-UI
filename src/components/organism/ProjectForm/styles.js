/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';
import theme from '../../../global/theme';

export const InputContainer = styled.div`
    border: 1px solid black;
    margin: 4px 0 8px 0;
    @media ${theme.media.tablet} {
      margin: 4px 0 16px 0;
    };
`;