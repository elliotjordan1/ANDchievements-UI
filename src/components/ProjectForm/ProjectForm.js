import React, { useState } from 'react';
import CreatableSelect from 'react-select/creatable';
import { Form, Label, Input, TextArea } from './styles';
import MultiSelect from '../MultiSelect';

const ProjectForm = () => {

    const clientOptions = [];
    const technologyOptions = [];
    const andiOptions = [];

    const clientList = [];
    const technologyList = [];
    const andiList = [];

    clientList.forEach(element => {
      const clientsValues = { value: element.value, label: element.label };
      clientOptions.push(clientsValues);
    });

    technologyList.forEach(element => {
      const technologiesValues = { value: element.value, label: element.label };
      technologyOptions.push(technologiesValues);
    });

    andiList.forEach(element => {
      const andiValues = { value: element.value, label: element.label };
      technologyOptions.push(andiValues);
    });

    const [projectTitle, setName] = useState('');
    const [projectDescription1, setProjectDescription1] = useState('');
    const [projectDescription2, setProjectDescription2] = useState('');
    const [projectDescription3, setProjectDescription3] = useState('');
    const [projectCoverImageUrl, setProjectCoverImageUrl] = useState('');

    return (
        <Form >
            <Label>
            Project Title:
            <Input
              type='text'
              value={projectTitle}
              onChange={e => setName(e.target.value)}
              placeholder='Enter your project title'
            />
            </Label>
            <Label>
            Client:
                <CreatableSelect
                options = {clientOptions}
                />
            </Label>
            <Label>
                About the client:
                <TextArea
                    type='text'
                    value={projectDescription1}
                    onChange={e => setProjectDescription1(e.target.value)}
                    placeholder='Enter client info'
                />
            </Label>
            <Label>
                Goals of the project:
                <TextArea
                    type='text'
                    value={projectDescription2}
                    onChange={e => setProjectDescription2(e.target.value)}
                    placeholder='Enter project goals'
                />
            </Label>
            <Label>
                End result of the project:
                <TextArea
                    type='text'
                    value={projectDescription3}
                    onChange={e => setProjectDescription3(e.target.value)}
                    placeholder='Enter project results'
                />
            </Label>
            <Label>
            Technology Stack:
                <CreatableSelect

                />
            </Label>

          <Label>
            select:
            <MultiSelect 
              placeholder="Select..."
            />
          </Label>


            <Label>
            Andis:
                <CreatableSelect
                      closeMenuOnSelect={false}
                      isMulti
                      options={andiOptions}
                />
            </Label>
            <Label>
                Project Cover Image URL:
                <Input
                  type='text'
                  value={projectCoverImageUrl}
                  onChange={e => setProjectCoverImageUrl(e.target.value)}
                  placeholder='Enter your project cover image URL'
                />
            </Label>
            <Label>
              <Input type='submit' value='Submit'/>
            </Label>
        </Form>
      );
}

export default ProjectForm;



