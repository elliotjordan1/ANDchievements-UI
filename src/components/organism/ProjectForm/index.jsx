import React, { useState } from 'react';
import { ToastProvider } from 'react-toast-notifications';
import {
  FormInput,
  FormLabel as Label,
  FormTitle,
  FormBody,
  TextArea,
  SubmitButton
} from '../../atom';
import {
  MultiSelect
} from '../../molecule';
import { FormWrapper, HomepageWrapper } from '../../../global/styles';
import * as AttributeTypes from '../../../global/constants';
import { InputContainer } from './styles';
import FormModal from '../FormModal';
import AttributeForm from '../AttributeForm';

const ProjectForm = () => { 
  const [viewModal, setViewModal] = useState(false);
  const [formType, setFormType] = useState();
  const [formTitle, setFormTitle] = useState();

  const setToANDi = () => {
    setFormType(<AttributeForm formType = {AttributeTypes.ANDi} />); 
    setFormTitle('Add an ANDi')
    setViewModal(true); 
  };

  const setToTechStack = () => {
    setFormType(<AttributeForm formType = {AttributeTypes.TechStack} />); 
    setFormTitle('Add a Tech Stack')
    setViewModal(true); 
  };

  const setToClient = () => {
    setFormType(<AttributeForm formType = {AttributeTypes.Client} />); 
    setFormTitle('Add a Client')
    setViewModal(true); 
  };
  
  return (
  <ToastProvider>
    <HomepageWrapper>
      <FormWrapper >
        <FormTitle text = "Add a Project" />
        <FormBody>
          <div>
            <Label labelText ="Project Title" />
            <FormInput maxLength={20} placeholder='Project title' />
          </div>
          <div>
            <Label onClick={() => {setToClient()}} labelText = "Client" />
            <FormInput placeholder='Select client' maxLength = {40} />
          </div>
          <div>
            <Label labelText = "Project Description" />
            <InputContainer>
              <TextArea maxLength={60} placeholder='Client description' />
              <TextArea maxLength={60} placeholder='Project description' />
              <TextArea maxLength={60} placeholder='Project outcomes' />
            </InputContainer>
          </div>
          <div>
            <Label labelText = "Cover Image" />
            <FormInput placeholder='Cover image url' maxLength={200}/>
          </div>
          <div>
            <Label onClick={() => {setToANDi()}} labelText = "ANDis" />
            <MultiSelect placeholder='Select ANDis...' />
          </div>
          <div>
            <Label onClick={() => {setToTechStack()}} labelText ="Tech Stacks" />
            <FormInput placeholder='Select Tech Stack' maxLength={40} />
          </div>
          <div>
            <SubmitButton text="SUBMIT" />
          </div>
        </FormBody>
      </FormWrapper>
      {
          (viewModal && (
            <FormModal 
              title={formTitle}
              form={formType}
              close={() => {setViewModal(false)}}
            /> 
          ))
          }
    </HomepageWrapper>
  </ToastProvider>
)};

export default ProjectForm;



