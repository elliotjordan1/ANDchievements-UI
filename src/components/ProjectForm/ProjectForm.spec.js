import React from 'react';
import { render } from '@testing-library/react';
import ProjectForm from './ProjectForm';

describe('ProjectForm component', () => {
  it('renders correctly', () => {
    const wrapper = render(<ProjectForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
