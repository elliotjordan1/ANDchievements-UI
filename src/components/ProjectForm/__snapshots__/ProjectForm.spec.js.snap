// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__HomepageWrapper-sc-101ii58-0 iCONwV"
      >
        <div
          class="styles__FormWrapper-sc-101ii58-3 foeRUl"
        >
          <div
            class="FormTitle__Title-oi1tko-0 dGHNUP"
          >
            Add a Project
          </div>
          <div
            class="FormBody__Body-sc-19652hi-0 kTauxf"
          >
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Project Title
              </div>
              <input
                class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
                maxlength="20"
                placeholder="Project title"
              />
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Client
              </div>
              <input
                class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
                maxlength="40"
                placeholder="Select client"
              />
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Project Description
              </div>
              <div
                class="styles__InputContainer-o9i1nk-0 dkIEjm"
              >
                <textarea
                  class="TextArea__Area-akp1wi-0 etxwJa"
                  maxlength="60"
                  placeholder="Client description"
                />
                <textarea
                  class="TextArea__Area-akp1wi-0 etxwJa"
                  maxlength="60"
                  placeholder="Project description"
                />
                <textarea
                  class="TextArea__Area-akp1wi-0 etxwJa"
                  maxlength="60"
                  placeholder="Project outcomes"
                />
              </div>
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Cover Image
              </div>
              <input
                class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
                maxlength="40"
                placeholder="Cover image url"
              />
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                ANDis
              </div>
              <input
                class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
                maxlength="40"
                placeholder="Select ANDis"
              />
            </div>
            <div>
              <div
                class="FormLabel__Label-wqrwj6-0 iJzIPZ"
              >
                Tech Stacks
              </div>
              <input
                class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
                maxlength="40"
                placeholder="Select Tech Stack"
              />
            </div>
            <div>
              <button
                class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="react-toast-notifications__container css-mrh2d-ToastContainer"
    />
  </body>,
  "container": <div>
    <div
      class="styles__HomepageWrapper-sc-101ii58-0 iCONwV"
    >
      <div
        class="styles__FormWrapper-sc-101ii58-3 foeRUl"
      >
        <div
          class="FormTitle__Title-oi1tko-0 dGHNUP"
        >
          Add a Project
        </div>
        <div
          class="FormBody__Body-sc-19652hi-0 kTauxf"
        >
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Project Title
            </div>
            <input
              class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
              maxlength="20"
              placeholder="Project title"
            />
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Client
            </div>
            <input
              class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
              maxlength="40"
              placeholder="Select client"
            />
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Project Description
            </div>
            <div
              class="styles__InputContainer-o9i1nk-0 dkIEjm"
            >
              <textarea
                class="TextArea__Area-akp1wi-0 etxwJa"
                maxlength="60"
                placeholder="Client description"
              />
              <textarea
                class="TextArea__Area-akp1wi-0 etxwJa"
                maxlength="60"
                placeholder="Project description"
              />
              <textarea
                class="TextArea__Area-akp1wi-0 etxwJa"
                maxlength="60"
                placeholder="Project outcomes"
              />
            </div>
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Cover Image
            </div>
            <input
              class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
              maxlength="40"
              placeholder="Cover image url"
            />
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              ANDis
            </div>
            <input
              class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
              maxlength="40"
              placeholder="Select ANDis"
            />
          </div>
          <div>
            <div
              class="FormLabel__Label-wqrwj6-0 iJzIPZ"
            >
              Tech Stacks
            </div>
            <input
              class="CustomFormInput__FormInput-zm333l-0 eLZzgK"
              maxlength="40"
              placeholder="Select Tech Stack"
            />
          </div>
          <div>
            <button
              class="SubmitButton__Button-sc-12s6sqe-0 bZVpPP"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
